tmt/checks/internal/timeout.py

```diff
@@ -0,0 +1,53 @@
+from typing import Optional
+
+import tmt.log
+import tmt.utils
+from tmt.checks import CheckPlugin, provides_check
+from tmt.checks.internal import InternalCheck
+from tmt.container import container
+from tmt.result import CheckResult, ResultOutcome
+from tmt.steps.execute import TestInvocation
+from tmt.utils import ProcessExitCodes
+
+CHECK_NAME = 'internal/timeout'
+
+
+@container
+class InternalTimeoutCheck(InternalCheck):
+    how: str = CHECK_NAME
+
+
+@provides_check(CHECK_NAME)
+class InternalTimeout(CheckPlugin[InternalTimeoutCheck]):
```

Comment:
I believe you don't need to include `Internal` in every class name, as they are already "marked" as internal by being in the `tmt.checks.internal` package.